app/models/form_profile.rb

```diff
@@ -232,40 +233,52 @@ def convert_vets360_address(address)
     }.compact
   end
 
-  def initialize_vets360_contact_info
+  def vets360_contact_info_hash
     return_val = {}
-    contact_information = Vet360Redis::ContactInformation.for_user(user)
-    return_val[:email] = contact_information.email&.email_address
+    return_val[:email] = vet360_contact_info&.email&.email_address
 
-    if contact_information.mailing_address.present?
-      return_val[:address] = convert_vets360_address(contact_information.mailing_address)
-    end
-    phone = contact_information.home_phone&.formatted_phone
+    return_val[:address] = vet360_mailing_address_hash if vet360_mailing_address.present?
+
+    phone = vet360_contact_info&.home_phone&.formatted_phone
     return_val[:us_phone] = phone
     return_val[:home_phone] = phone
-    return_val[:mobile_phone] = contact_information.mobile_phone&.formatted_phone
+    return_val[:mobile_phone] = vet360_contact_info&.mobile_phone&.formatted_phone
 
     return_val
   end
 
   def initialize_contact_information
     opt = {}
-    opt.merge!(initialize_vets360_contact_info) if Settings.vet360.prefill && user.vet360_id.present?
+    opt.merge!(vets360_contact_info_hash) if vet360_contact_info
 
     opt[:address] ||= va_profile_address_hash
 
     opt[:email] ||= extract_pciu_data(:pciu_email)
     if opt[:home_phone].nil?
-      pciu_primary_phone = extract_pciu_data(:pciu_primary_phone)
       opt[:home_phone] = pciu_primary_phone
-      opt[:us_phone] = get_us_phone(pciu_primary_phone)
+      opt[:us_phone] = pciu_us_phone
     end
 
     format_for_schema_compatibility(opt)
 
     FormContactInformation.new(opt)
   end
 
+  # doing this (below) instead of `@vet360_contact_info ||= Settings...` to cache nil too
+  def vet360_contact_info
+    return @vet360_contact_info if @vet360_contact_info_retrieved
+
+    @vet360_contact_info_retrieved = true
+    @vet360_contact_info = Settings.vet360.prefill &&
```

Comment:
I think this is confusing because it reads like a boolean statement but it's actually setting a value when other values are present.  I think it would be easier to read if it were something like this:
```
  def vet360_contact_info
    return @vet360_contact_info if @vet360_contact_info_retrieved

    @vet360_contact_info_retrieved = true
    if Settings.vet360.prefill && user.vet360_id.present?
      @vet360_contact_info = Vet360Redis::ContactInformation.for_user(user)
    end
  end
```

Reply:
i like it! much easier to grok 💯

Reply:
@erluti do you think it'd be good to add this final line, just to make it more clear that `@vet360_contact_info` will be returned? 
```
  def vet360_contact_info
    return @vet360_contact_info if @vet360_contact_info_retrieved

    @vet360_contact_info_retrieved = true
    if Settings.vet360.prefill && user.vet360_id.present?
      @vet360_contact_info = Vet360Redis::ContactInformation.for_user(user)
    end
    @vet360_contact_info # <-- HERE
  end
```

Reply:
i'm always a fan of clarity!